lib/neo4j/core/cypher_session/adaptors/bolt.rb

```diff
@@ -217,14 +217,29 @@ def flush_response
             if !(header = recvmsg(2)).empty? && (chunk_size = header.unpack('s>*')[0]) > 0
               log_message :S, :chunk_size, chunk_size
 
-              chunk = recvmsg(chunk_size)
+              chunk = flush_chunks(chunk_size)
 
               unpacker = PackStream::Unpacker.new(StringIO.new(chunk))
 
               [].tap { |r| while arg = unpacker.unpack_value!; r << arg; end }
             end
           end
 
+          def flush_chunks(first_chunk_size)
+            chunk_size = first_chunk_size
+            chunk = ''
+
+            loop do
+              chunk += recvmsg(chunk_size)
+              chunk_size = recvmsg(2, true).unpack('s>*')[0]
+
+              break if chunk_size.zero?
+              recvmsg(2)
+            end
+
+            chunk
+          end
```

Comment:
I think that you could do something like move `flush_chunks` before the `if` on line 217 which would mean you only have one place where we are doing `recvmsg(2).unpack('s>*')[0]`.  That would mean we wouldn't need to pass in `first_chunk_size` and we could perhaps have a nice `while` look instead of just a plain `loop`.